web/source/osk/visualKeyboard.ts

```diff
@@ -1675,26 +1576,125 @@ namespace com.keyman.osk {
+     * @returns 
+     */
+    startLongpress(key: KeyElement): PendingGesture {
+      let _this = this;
+
+      // First-level object/Promise:  will produce a subkey popup when the longpress gesture completes.
+      // 'Returns' a second-level object/Promise:  resolves when a subkey is selected or is cancelled.
+      let pendingLongpress = new browser.PendingLongpress(this, key);
+      pendingLongpress.promise.then(function(subkeyPopup) {
+        // In-browser-specific handling.
+        if(subkeyPopup) {
+          // Append the touch-hold (subkey) array to the OSK
+          let keyman = com.keyman.singleton;
+          keyman.osk._Box.appendChild(subkeyPopup.element);
+          keyman.osk._Box.appendChild(subkeyPopup.shim);
+
+          // Must be placed after its `.element` has been inserted into the DOM.
+          subkeyPopup.reposition(_this);
+        }
+      });
+
+      return pendingLongpress;
     }
 
-    if(typeof key['subKeys'] != 'undefined' && key['subKeys'] != null) {
-      this.subkeyDelayTimer = window.setTimeout(
-        function(this: VisualKeyboard) {
-          this.clearPopup();
-          this.showSubKeys(key);
-        }.bind(this), this.popupDelay);
+    /**
+     * Initializes all supported gestures given a base key and the triggering touch coordinates.
+     * @param key     The gesture's base key
+     * @param touch   The starting touch coordinates for the gesture
+     * @returns 
+     */
+    initGestures(key: KeyElement, touch: Touch) {
+      if(key['subKeys']) {
+        let _this = this;
+
+        let pendingLongpress = this.startLongpress(key);
+        if(pendingLongpress == null) {
+          return;
+        }
+        this.pendingSubkey = pendingLongpress;
+
+        pendingLongpress.promise.then(function(subkeyPopup) {
+          if(_this.pendingSubkey == pendingLongpress) {
+            _this.pendingSubkey = null;
+          }
+
+          if(subkeyPopup) {
+            // Clear key preview if any
+            _this.showKeyTip(null,false);
+  
+            _this.subkeyGesture = subkeyPopup;
+            subkeyPopup.promise.then(function(keyEvent: text.KeyEvent) {
+              // Allow active cancellation, even if the source should allow passive.
+              // It's an easy and cheap null guard.
+              if(keyEvent) {
+                PreProcessor.raiseKeyEvent(keyEvent);
+              }
+              _this.clearPopup();
+            });
+          }
+        });
+      }
+    }
+
+    /**
+     * Updates all currently-pending and activated gestures.
+     * 
+     * @param currentKey    The key currently underneath the most recent touch coordinate
+     * @param previousKey   The previously-selected key
+     * @param touch         The current touch-coordinate for the gesture
+     * @returns true if should fully capture input, false if input should 'fall through'.
+     */
+    updateGestures(currentKey: KeyElement, previousKey: KeyElement, touch: Touch): boolean {
```

Comment:
Centralizes the update-related aspects originally within the `moveOver` event handler.